pkg/client/events_query/connection.go

```diff
@@ -0,0 +1,34 @@
+package eventsquery
+
+import (
+	"github.com/gorilla/websocket"
+
+	"pocket/pkg/client"
+)
+
+var _ client.Connection = (*websocketConn)(nil)
+
+// websocketConn implements the Connection interface using the gorilla websocket
+// transport implementation.
+type websocketConn struct {
+	conn *websocket.Conn
+}
+
+// Receive implements the respective interface method using the underlying websocket.
+func (wsConn *websocketConn) Receive() ([]byte, error) {
+	_, msg, err := wsConn.conn.ReadMessage()
+	if err != nil {
+		return nil, ErrReceive.Wrapf("%s", err)
+	}
+	return msg, nil
+}
+
+// Send implements the respective interface method using the underlying websocket.
+func (wsConn *websocketConn) Send(msg []byte) error {
+	return wsConn.conn.WriteMessage(websocket.TextMessage, msg)
```

Comment:
Why is this a `TextMessage` and not a `ByteMessage`?

Reply:
IIRC, this is the gorilla websocket client implementation exposing a websocket degree of freedom which has to do with the encoding (or something along those lines). Since we're working with a JSON-RPC API, the encoding is text, not binary.

Reply:
(I describe what happens when we try to use binary messages in https://github.com/pokt-network/poktroll/pull/64#discussion_r1372247584)

Reply:
#PUC